tests/integration/test_metrics.py

```diff
@@ -919,11 +947,13 @@ def test_limit_custom_measurements(
     event, _ = transactions_consumer.get_event()
     assert len(event["measurements"]) == 2
 
-    # Expect exactly 3 metrics (transaction.duration, 1 builtin, 1 custom)
-    metrics = metrics_by_name(metrics_consumer, 3)
+    # Expect exactly 4 metrics:
+    # (transaction.duration, transactions.count_per_root_project, 1 builtin, 1 custom)
+    metrics = metrics_by_name(metrics_consumer, 4)
 
     assert metrics.keys() == {
         "d:transactions/duration@millisecond",
+        "c:transactions/count_per_root_project@none",
```

Comment:
As far as I can tell, these test cases all add this metric to current project, because there is no DSC involved. Please also add tests for the following:

* a test with a DSC where the DSC's project ID is different from the event's project ID.
* a test where we assert that the DSC's `transaction` field is used as a tag on the metric.